File: ActiveX Controls/ASTC 1.50/Advanced Sytem Tray Control/Samples/VB.NET 2003/TrayIcon Sample/Form1.vb
'=========================================================================================
'Publisher      CprinGold Software.
'               http://www.cpringold.com
'               support@cpringold.com
'
'
' Description:  A sample code demonstrate how to add your software icon to system tray area
'               As well as how to use the advanced features related to system tray area.
'==========================================================================================
'ASTC provide you with a very simple object, TrayIcon object which is the responsible to add the most of system
'Tray functions to your applications.
'
'With a few lines i can demonstrate how TrayIcon object can help you
'
'Use AxTrayIcon1.Show method to add an icon to system tray area
'Use AxTrayIcon1.Hide method to hide the icon you had add by TrayIcon.Show method from system tray
'Use AxTrayIcon1. ChangeIcon method to change the icon you had added by another icon
'Use AxTrayIcon1.Animate method to animate the icon in the system tray
'Use AxTrayIcon1.StopAnimateing method to stop animating the icon
'Use AxTrayIcon1.ShowBalloon to display a tooltip balloon
'Use AxTrayIcon1.HideBalloon to hide the balloon
'Use AxTrayIcon1.Popup to display a popup menu
'Use AxTrayIcon1.TrackPopupmenu to track the popup menu whenever it lost focus in order to close it.
'Use AxTrayIcon1.IsDisplayed to check if the icon is displayed in the system tray area
'
'Note that you have a custom types of events associated with TrayIcon Object like
'BalloonShow, BalloonHide, MedMouseUp, LeftMouseUp, RightMouseUp, MedMouseDown, LeftMouseDown
'RightMouseDown, MedMouseDBLCLK, LeftMouseDBLCLK, BalloonLeftClick, BalloonRightClick, RightMouseDBLCLK ()
'
'Note that TrayIcon Object doesn't subclass the window in which it contains TrayIcon control
'because TrayIcon Object dynamically creates an internal window related to it's instance,
'and destroy this window whenever you call Hide method or close your application,
'so don 't worry about subclassing
'If you have any further questions or need more sample code don't hesitate to contact us
'At support@cpringold.com

Public Class Form1
    Inherits System.Windows.Forms.Form

#Region " Windows Form Designer generated code "

    Public Sub New()
        MyBase.New()

        'This call is required by the Windows Form Designer.
        InitializeComponent()

        'Add any initialization after the InitializeComponent() call

    End Sub

    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    Friend WithEvents groupBox1 As System.Windows.Forms.GroupBox
    Friend WithEvents AxImageList1 As AxMSComctlLib.AxImageList
    Friend WithEvents butSTrack As System.Windows.Forms.Button
    Friend WithEvents butTrack As System.Windows.Forms.Button
    Friend WithEvents butCBalloon As System.Windows.Forms.Button
    Friend WithEvents butShowBalloon As System.Windows.Forms.Button
    Friend WithEvents butStAni As System.Windows.Forms.Button
    Friend WithEvents butAni As System.Windows.Forms.Button
    Friend WithEvents butHide As System.Windows.Forms.Button
    Friend WithEvents butChgIcon As System.Windows.Forms.Button
    Friend WithEvents butShow As System.Windows.Forms.Button

    Friend WithEvents ContextMenu1 As System.Windows.Forms.ContextMenu
    Friend WithEvents MenuItem1 As System.Windows.Forms.MenuItem
    Friend WithEvents MenuItem2 As System.Windows.Forms.MenuItem
    Friend WithEvents AxTrayIcon1 As AxASTC.AxTrayIcon
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Dim resources As System.Resources.ResourceManager = New System.Resources.ResourceManager(GetType(Form1))
        Me.groupBox1 = New System.Windows.Forms.GroupBox
        Me.AxImageList1 = New AxMSComctlLib.AxImageList
        Me.butSTrack = New System.Windows.Forms.Button
        Me.butTrack = New System.Windows.Forms.Button
        Me.butCBalloon = New System.Windows.Forms.Button
        Me.butShowBalloon = New System.Windows.Forms.Button
        Me.butStAni = New System.Windows.Forms.Button
        Me.butAni = New System.Windows.Forms.Button
        Me.butHide = New System.Windows.Forms.Button
        Me.butChgIcon = New System.Windows.Forms.Button
        Me.butShow = New System.Windows.Forms.Button
        Me.ContextMenu1 = New System.Windows.Forms.ContextMenu
        Me.MenuItem1 = New System.Windows.Forms.MenuItem
        Me.MenuItem2 = New System.Windows.Forms.MenuItem
        Me.AxTrayIcon1 = New AxASTC.AxTrayIcon
        Me.groupBox1.SuspendLayout()
        CType(Me.AxImageList1, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.AxTrayIcon1, System.ComponentModel.ISupportInitialize).BeginInit()
        Me.SuspendLayout()
        '
        'groupBox1
        '
        Me.groupBox1.Controls.Add(Me.AxTrayIcon1)
        Me.groupBox1.Controls.Add(Me.AxImageList1)
        Me.groupBox1.Controls.Add(Me.butSTrack)
        Me.groupBox1.Controls.Add(Me.butTrack)
        Me.groupBox1.Controls.Add(Me.butCBalloon)
        Me.groupBox1.Controls.Add(Me.butShowBalloon)
        Me.groupBox1.Controls.Add(Me.butStAni)
        Me.groupBox1.Controls.Add(Me.butAni)
        Me.groupBox1.Controls.Add(Me.butHide)
        Me.groupBox1.Controls.Add(Me.butChgIcon)
        Me.groupBox1.Controls.Add(Me.butShow)
        Me.groupBox1.Location = New System.Drawing.Point(8, 8)
        Me.groupBox1.Name = "groupBox1"
        Me.groupBox1.Size = New System.Drawing.Size(216, 176)
        Me.groupBox1.TabIndex = 6
        Me.groupBox1.TabStop = False
        '
        'AxImageList1
        '
        Me.AxImageList1.ContainingControl = Me
        Me.AxImageList1.Enabled = True
        Me.AxImageList1.Location = New System.Drawing.Point(88, 112)
        Me.AxImageList1.Name = "AxImageList1"
        Me.AxImageList1.OcxState = CType(resources.GetObject("AxImageList1.OcxState"), System.Windows.Forms.AxHost.State)
        Me.AxImageList1.Size = New System.Drawing.Size(38, 38)
        Me.AxImageList1.TabIndex = 18
        '
        'butSTrack
        '
        Me.butSTrack.Enabled = False
        Me.butSTrack.Location = New System.Drawing.Point(112, 112)
        Me.butSTrack.Name = "butSTrack"
        Me.butSTrack.Size = New System.Drawing.Size(96, 24)
        Me.butSTrack.TabIndex = 14
        Me.butSTrack.Text = "Stop Track"
        '
        'butTrack
        '
        Me.butTrack.Location = New System.Drawing.Point(112, 80)
        Me.butTrack.Name = "butTrack"
        Me.butTrack.Size = New System.Drawing.Size(96, 24)
        Me.butTrack.TabIndex = 13
        Me.butTrack.Text = "TrackPopUp"
        '
        'butCBalloon
        '
        Me.butCBalloon.Location = New System.Drawing.Point(112, 48)
        Me.butCBalloon.Name = "butCBalloon"
        Me.butCBalloon.Size = New System.Drawing.Size(96, 24)
        Me.butCBalloon.TabIndex = 12
        Me.butCBalloon.Text = "Close Balloon"
        '
        'butShowBalloon
        '
        Me.butShowBalloon.Location = New System.Drawing.Point(112, 16)
        Me.butShowBalloon.Name = "butShowBalloon"
        Me.butShowBalloon.Size = New System.Drawing.Size(96, 24)
        Me.butShowBalloon.TabIndex = 11
        Me.butShowBalloon.Text = "Show Balloon"
        '
        'butStAni
        '
        Me.butStAni.Location = New System.Drawing.Point(56, 144)
        Me.butStAni.Name = "butStAni"
        Me.butStAni.Size = New System.Drawing.Size(104, 24)
        Me.butStAni.TabIndex = 9
        Me.butStAni.Text = "Stop Animation"
        '
        'butAni
        '
        Me.butAni.Location = New System.Drawing.Point(8, 112)
        Me.butAni.Name = "butAni"
        Me.butAni.Size = New System.Drawing.Size(96, 24)
        Me.butAni.TabIndex = 7
        Me.butAni.Text = "Animate"
        '
        'butHide
        '
        Me.butHide.Location = New System.Drawing.Point(8, 80)
        Me.butHide.Name = "butHide"
        Me.butHide.Size = New System.Drawing.Size(96, 24)
        Me.butHide.TabIndex = 6
        Me.butHide.Text = "Hide Icon"
        '
        'butChgIcon
        '
        Me.butChgIcon.Location = New System.Drawing.Point(8, 48)
        Me.butChgIcon.Name = "butChgIcon"
        Me.butChgIcon.Size = New System.Drawing.Size(96, 24)
        Me.butChgIcon.TabIndex = 4
        Me.butChgIcon.Text = "Change Icon"
        '
        'butShow
        '
        Me.butShow.Location = New System.Drawing.Point(8, 16)
        Me.butShow.Name = "butShow"
        Me.butShow.Size = New System.Drawing.Size(96, 24)
        Me.butShow.TabIndex = 2
        Me.butShow.Text = "&Show Icon"
        '
        'ContextMenu1
        '
        Me.ContextMenu1.MenuItems.AddRange(New System.Windows.Forms.MenuItem() {Me.MenuItem1, Me.MenuItem2})
        '
        'MenuItem1
        '
        Me.MenuItem1.Index = 0
        Me.MenuItem1.Text = "main"
        '
        'MenuItem2
        '
        Me.MenuItem2.Index = 1
        Me.MenuItem2.Text = "Exit"
        '
        'AxTrayIcon1
        '
        Me.AxTrayIcon1.ContainingControl = Me
        Me.AxTrayIcon1.Enabled = True
        Me.AxTrayIcon1.Location = New System.Drawing.Point(88, 72)
        Me.AxTrayIcon1.Name = "AxTrayIcon1"
        Me.AxTrayIcon1.OcxState = CType(resources.GetObject("AxTrayIcon1.OcxState"), System.Windows.Forms.AxHost.State)
        Me.AxTrayIcon1.Size = New System.Drawing.Size(36, 36)
        Me.AxTrayIcon1.TabIndex = 19
        '
        'Form1
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(232, 190)
        Me.Controls.Add(Me.groupBox1)
        Me.Name = "Form1"
        Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen
        Me.Text = "VB.NET TrayIcon"
        Me.groupBox1.ResumeLayout(False)
        CType(Me.AxImageList1, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.AxTrayIcon1, System.ComponentModel.ISupportInitialize).EndInit()
        Me.ResumeLayout(False)

    End Sub

#End Region
    '==============================================================================
    ' Method:        butShow_Click
    '
    ' Description:  Shows the icon in system tray.
    '==============================================================================
    Private Sub butShow_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butShow.Click
        'Unlike the previous version of ASTC, you don't need to pass a window handle to Show function because
        'ASTC dynamically creates an internal window related to ASTC instance, and destroy this window whenever
        'you call Hide method or close your application.
        AxTrayIcon1().CtlShow(Me.Icon.Handle.ToInt32(), "VB.netTrayIcon")
    End Sub
    '==============================================================================
    ' Method:        butChgIcon_Click
    '
    ' Description:  Changes the icon in system tray area.
    '==============================================================================
    Private Sub butChgIcon_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butChgIcon.Click
        'change the icon in system tray area
        'by another one in ImageList    
        AxTrayIcon1.ChangeIcon(AxImageList1.ListImages.Item(2).ExtractIcon().Handle)
    End Sub
    '==============================================================================
    ' Method:       butHide_Click
    '
    ' Description:  Removes the icon from system tray.
    '==============================================================================
    Private Sub butHide_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butHide.Click
        'Hide the icon Only if it was displayed
        If AxTrayIcon1.IsDisplayed Then AxTrayIcon1.CtlHide()
    End Sub
    '==================================================================================
    ' Method:        butAni_Click
    '
    ' Description:  Animate the icon in system tray using in icons in imagelist control.
    '==================================================================================
    Private Sub butAni_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butAni.Click
        'only if the icon is displayed
        If AxTrayIcon1.IsDisplayed Then
            'Only if the icon is not Animated   
            If AxTrayIcon1.AnimateState = False Then
                'Start animating the icons in system tray
                'we use the icons in the imagelist control
                'Then
                'We pass the ImageList1 control to Animate method
                '''you can replace the icons with your own'''
                AxTrayIcon1.Animate(AxImageList1, 1)
            End If
        End If
    End Sub
    '==============================================================================
    ' Method:       butStAni_Click
    '
    ' Description:  Stop animating the icons in system tray.
    '==============================================================================
    Private Sub butStAni_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butStAni.Click
        'Only if the icon is animated 
        If AxTrayIcon1.AnimateState = True Then
            AxTrayIcon1.StopAnimateing(Me.Icon.Handle.ToInt32())
        End If
    End Sub
    '==============================================================================
    ' Method:       butShowBalloon_Click
    '
    ' Description:  Displays a balloon tooltip points to the icon.
    '==============================================================================
    Private Sub butShowBalloon_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butShowBalloon.Click
        If AxTrayIcon1.AnimateState = True Then MsgBox("You can't Display a balloon while animating the icon.")
        AxTrayIcon1.ShowBalloon("Click me", "VB.net TrayIcon", ASTC.BIcoType.Info, 5000)
    End Sub
    '==============================================================================
    ' Method:       butCBalloon_Click
    '
    ' Description:  Close the balloon tooltip.
    '==============================================================================
    Private Sub butCBalloon_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butCBalloon.Click
        AxTrayIcon1.CloseBalloon()
    End Sub
    '==============================================================================
    ' Method:       butTrack_Click
    '
    ' Description:  Start Tracking the popup menu in system tray area.
    '==============================================================================
    Private Sub butTrack_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butTrack.Click
        MsgBox("when you right click the icon a popupmenu will appear,ASTC will Track the popupmenu and close it when you left click again")
        'Track the popup menu so ASTC can destroy it whenever it lost focus   
        AxTrayIcon1.TrackPopMenu = True
        butSTrack.Enabled = True
    End Sub
    '==============================================================================
    ' Method:       butSTrack_Click
    '
    ' Description:  Stop track the popup menu in system tray area.
    '==============================================================================
    Private Sub butSTrack_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles butSTrack.Click
        MsgBox("ASTC stop tracking the popupmenu,Now right click the icon again and you will see that popupmenu willn't close unless you click an item.")
        'stop tracking the popup menu
        AxTrayIcon1.TrackPopMenu = False
        butSTrack.Enabled = False
    End Sub
    '==============================================================================
    ' Method:       Form1_Closing
    '
    ' Description:  Removes the icon from system tray whenever the
    '               terminates the program.
    '==============================================================================
    Private Sub Form1_Closing(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs) Handles MyBase.Closing
        If AxTrayIcon1.IsDisplayed Then AxTrayIcon1.CtlHide()
    End Sub
    '==============================================================================
    ' Method:        MenuItem2_Click
    '
    ' Description:  Removes the icon from system tray.
    '==============================================================================
    Private Sub MenuItem2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MenuItem2.Click
        If AxTrayIcon1.IsDisplayed Then AxTrayIcon1.CtlHide()
        End
    End Sub
    Private Sub MenuItem1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MenuItem1.Click
        MsgBox("main menu Item had been clicked.")
    End Sub
    '====================================================================================
    ' Method:       AxTrayIcon1_RightMouseUp
    '
    ' Description:  Display a popup menu whenever you right click the icon in system tray.
    '====================================================================================
    Private Sub AxTrayIcon1_RightMouseUp(ByVal sender As Object, ByVal e As System.EventArgs) Handles AxTrayIcon1.RightMouseUp
        AxTrayIcon1.PopUp(ContextMenu1.Handle.ToInt32(), Handle.ToInt32())
    End Sub
    '==============================================================================
    ' Method:       AxTrayIcon1_BalloonLeftClick
    '
    ' Description:  Display a message box whenever you left click the balloon tooltip.
    '==============================================================================
    Private Sub AxTrayIcon1_BalloonLeftClick(ByVal sender As Object, ByVal e As System.EventArgs) Handles AxTrayIcon1.BalloonLeftClick
        MsgBox("The Balloon Tooltip had been left clicked.")
    End Sub
End Class

File: ActiveX Controls/ASTC 1.50/Advanced Sytem Tray Control/Samples/VC#.NET 2003/TrayIcon Sample/Form1.cs
/*
=========================================================================================
Publisher      CprinGold Software.
               http://www.cpringold.com
               support@cpringold.com


 Description:  A sample code demonstrate how to add your software icon to system tray area
               As well as how to use the advanced features related to system tray area.
==========================================================================================
ASTC provide you with a very simple object, TrayIcon object which is the responsible to add the most of system
Tray functions to your applications.

With a few lines i can demonstrate how TrayIcon object can help you

Use AxTrayIcon1.Show method to add an icon to system tray area
Use AxTrayIcon1.Hide method to hide the icon you had add by TrayIcon.Show method from system tray
Use AxTrayIcon1. ChangeIcon method to change the icon you had added by another icon
Use AxTrayIcon1.Animate method to animate the icon in the system tray
Use AxTrayIcon1.StopAnimateing method to stop animating the icon
Use AxTrayIcon1.ShowBalloon to display a tooltip balloon
Use AxTrayIcon1.HideBalloon to hide the balloon
Use AxTrayIcon1.Popup to display a popup menu
Use AxTrayIcon1.TrackPopupmenu to track the popup menu whenever it lost focus in order to close it.
Use AxTrayIcon1.IsDisplayed to check if the icon is displayed in the system tray area

Note that you have a custom types of events associated with TrayIcon Object like
BalloonShow, BalloonHide, MedMouseUp, LeftMouseUp, RightMouseUp, MedMouseDown, LeftMouseDown
RightMouseDown, MedMouseDBLCLK, LeftMouseDBLCLK, BalloonLeftClick, BalloonRightClick, RightMouseDBLCLK ()

Note that TrayIcon Object doesn't subclass the window in which it contains TrayIcon control
because TrayIcon Object dynamically creates an internal window related to it's instance,
and destroy this window whenever you call Hide method or close your application,
so don 't worry about subclassing
If you have any further questions or need more sample code don't hesitate to contact us
At support@cpringold.com
*/
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace TrayIcon_Sample
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
		internal System.Windows.Forms.GroupBox groupBox1;
		internal System.Windows.Forms.Button butSTrack;
		internal System.Windows.Forms.Button butTrack;
		internal System.Windows.Forms.Button butCBalloon;
		internal System.Windows.Forms.Button butShowBalloon;
		internal System.Windows.Forms.Button butStAni;
		internal System.Windows.Forms.Button butAni;
		internal System.Windows.Forms.Button butHide;
		internal System.Windows.Forms.Button butChgIcon;
		internal System.Windows.Forms.Button butShow;
		private AxMSComctlLib.AxImageList axImageList1;
		internal System.Windows.Forms.ContextMenu ContextMenu1;
		internal System.Windows.Forms.MenuItem MenuItem1;
		internal System.Windows.Forms.MenuItem MenuItem2;
		private AxASTC.AxTrayIcon axTrayIcon1;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			System.Resources.ResourceManager resources = new System.Resources.ResourceManager(typeof(Form1));
			this.groupBox1 = new System.Windows.Forms.GroupBox();
			this.axImageList1 = new AxMSComctlLib.AxImageList();
			this.butSTrack = new System.Windows.Forms.Button();
			this.butTrack = new System.Windows.Forms.Button();
			this.butCBalloon = new System.Windows.Forms.Button();
			this.butShowBalloon = new System.Windows.Forms.Button();
			this.butStAni = new System.Windows.Forms.Button();
			this.butAni = new System.Windows.Forms.Button();
			this.butHide = new System.Windows.Forms.Button();
			this.butChgIcon = new System.Windows.Forms.Button();
			this.butShow = new System.Windows.Forms.Button();
			this.ContextMenu1 = new System.Windows.Forms.ContextMenu();
			this.MenuItem1 = new System.Windows.Forms.MenuItem();
			this.MenuItem2 = new System.Windows.Forms.MenuItem();
			this.axTrayIcon1 = new AxASTC.AxTrayIcon();
			this.groupBox1.SuspendLayout();
			((System.ComponentModel.ISupportInitialize)(this.axImageList1)).BeginInit();
			((System.ComponentModel.ISupportInitialize)(this.axTrayIcon1)).BeginInit();
			this.SuspendLayout();
			// 
			// groupBox1
			// 
			this.groupBox1.Controls.Add(this.axTrayIcon1);
			this.groupBox1.Controls.Add(this.axImageList1);
			this.groupBox1.Controls.Add(this.butSTrack);
			this.groupBox1.Controls.Add(this.butTrack);
			this.groupBox1.Controls.Add(this.butCBalloon);
			this.groupBox1.Controls.Add(this.butShowBalloon);
			this.groupBox1.Controls.Add(this.butStAni);
			this.groupBox1.Controls.Add(this.butAni);
			this.groupBox1.Controls.Add(this.butHide);
			this.groupBox1.Controls.Add(this.butChgIcon);
			this.groupBox1.Controls.Add(this.butShow);
			this.groupBox1.Location = new System.Drawing.Point(8, 8);
			this.groupBox1.Name = "groupBox1";
			this.groupBox1.Size = new System.Drawing.Size(216, 176);
			this.groupBox1.TabIndex = 7;
			this.groupBox1.TabStop = false;
			// 
			// axImageList1
			// 
			this.axImageList1.ContainingControl = this;
			this.axImageList1.Enabled = true;
			this.axImageList1.Location = new System.Drawing.Point(8, 136);
			this.axImageList1.Name = "axImageList1";
			this.axImageList1.OcxState = ((System.Windows.Forms.AxHost.State)(resources.GetObject("axImageList1.OcxState")));
			this.axImageList1.Size = new System.Drawing.Size(38, 38);
			this.axImageList1.TabIndex = 15;
			// 
			// butSTrack
			// 
			this.butSTrack.Enabled = false;
			this.butSTrack.Location = new System.Drawing.Point(112, 112);
			this.butSTrack.Name = "butSTrack";
			this.butSTrack.Size = new System.Drawing.Size(96, 24);
			this.butSTrack.TabIndex = 14;
			this.butSTrack.Text = "Stop Track";
			this.butSTrack.Click += new System.EventHandler(this.butSTrack_Click);
			// 
			// butTrack
			// 
			this.butTrack.Location = new System.Drawing.Point(112, 80);
			this.butTrack.Name = "butTrack";
			this.butTrack.Size = new System.Drawing.Size(96, 24);
			this.butTrack.TabIndex = 13;
			this.butTrack.Text = "TrackPopUp";
			this.butTrack.Click += new System.EventHandler(this.butTrack_Click);
			// 
			// butCBalloon
			// 
			this.butCBalloon.Location = new System.Drawing.Point(112, 48);
			this.butCBalloon.Name = "butCBalloon";
			this.butCBalloon.Size = new System.Drawing.Size(96, 24);
			this.butCBalloon.TabIndex = 12;
			this.butCBalloon.Text = "Close Balloon";
			this.butCBalloon.Click += new System.EventHandler(this.butCBalloon_Click);
			// 
			// butShowBalloon
			// 
			this.butShowBalloon.Location = new System.Drawing.Point(112, 16);
			this.butShowBalloon.Name = "butShowBalloon";
			this.butShowBalloon.Size = new System.Drawing.Size(96, 24);
			this.butShowBalloon.TabIndex = 11;
			this.butShowBalloon.Text = "Show Balloon";
			this.butShowBalloon.Click += new System.EventHandler(this.butShowBalloon_Click);
			// 
			// butStAni
			// 
			this.butStAni.Location = new System.Drawing.Point(56, 144);
			this.butStAni.Name = "butStAni";
			this.butStAni.Size = new System.Drawing.Size(104, 24);
			this.butStAni.TabIndex = 9;
			this.butStAni.Text = "Stop Animation";
			this.butStAni.Click += new System.EventHandler(this.butStAni_Click);
			// 
			// butAni
			// 
			this.butAni.Location = new System.Drawing.Point(8, 112);
			this.butAni.Name = "butAni";
			this.butAni.Size = new System.Drawing.Size(96, 24);
			this.butAni.TabIndex = 7;
			this.butAni.Text = "Animate";
			this.butAni.Click += new System.EventHandler(this.butAni_Click);
			// 
			// butHide
			// 
			this.butHide.Location = new System.Drawing.Point(8, 80);
			this.butHide.Name = "butHide";
			this.butHide.Size = new System.Drawing.Size(96, 24);
			this.butHide.TabIndex = 6;
			this.butHide.Text = "Hide Icon";
			this.butHide.Click += new System.EventHandler(this.butHide_Click);
			// 
			// butChgIcon
			// 
			this.butChgIcon.Location = new System.Drawing.Point(8, 48);
			this.butChgIcon.Name = "butChgIcon";
			this.butChgIcon.Size = new System.Drawing.Size(96, 24);
			this.butChgIcon.TabIndex = 4;
			this.butChgIcon.Text = "Change Icon";
			this.butChgIcon.Click += new System.EventHandler(this.butChgIcon_Click);
			// 
			// butShow
			// 
			this.butShow.Location = new System.Drawing.Point(8, 16);
			this.butShow.Name = "butShow";
			this.butShow.Size = new System.Drawing.Size(96, 24);
			this.butShow.TabIndex = 2;
			this.butShow.Text = "&Show Icon";
			this.butShow.Click += new System.EventHandler(this.butShow_Click);
			// 
			// ContextMenu1
			// 
			this.ContextMenu1.MenuItems.AddRange(new System.Windows.Forms.MenuItem[] {
																						 this.MenuItem1,
																						 this.MenuItem2});
			// 
			// MenuItem1
			// 
			this.MenuItem1.Index = 0;
			this.MenuItem1.Text = "main";
			this.MenuItem1.Click += new System.EventHandler(this.MenuItem1_Click);
			// 
			// MenuItem2
			// 
			this.MenuItem2.Index = 1;
			this.MenuItem2.Text = "Exit";
			this.MenuItem2.Click += new System.EventHandler(this.MenuItem2_Click);
			// 
			// axTrayIcon1
			// 
			this.axTrayIcon1.ContainingControl = this;
			this.axTrayIcon1.Enabled = true;
			this.axTrayIcon1.Location = new System.Drawing.Point(88, 72);
			this.axTrayIcon1.Name = "axTrayIcon1";
			this.axTrayIcon1.OcxState = ((System.Windows.Forms.AxHost.State)(resources.GetObject("axTrayIcon1.OcxState")));
			this.axTrayIcon1.Size = new System.Drawing.Size(36, 36);
			this.axTrayIcon1.TabIndex = 16;
			this.axTrayIcon1.RightMouseUp += new System.EventHandler(this.axTrayIcon1_RightMouseUp);
			this.axTrayIcon1.BalloonLeftClick += new System.EventHandler(this.axTrayIcon1_BalloonLeftClick);
			// 
			// Form1
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(232, 190);
			this.Controls.Add(this.groupBox1);
			this.Name = "Form1";
			this.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen;
			this.Text = "VC#.NET TrayIcon Sample Code";
			this.Closing += new System.ComponentModel.CancelEventHandler(this.Form1_Closing);
			this.groupBox1.ResumeLayout(false);
			((System.ComponentModel.ISupportInitialize)(this.axImageList1)).EndInit();
			((System.ComponentModel.ISupportInitialize)(this.axTrayIcon1)).EndInit();
			this.ResumeLayout(false);

		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		/*    
		==============================================================================
		 Method:        butShow_Click
		
		 Description:  Shows the icon in system tray.
		==============================================================================
        */
		private void butShow_Click(object sender, System.EventArgs e)
		{
		 /*
		    Unlike the previous version of ASTC, you don't need to pass a window handle to Show function because
            ASTC dynamically creates an internal window related to ASTC instance, and destroy this window whenever
            you call Hide method or close your application.
		 */
			axTrayIcon1.CtlShow(this.Icon.Handle.ToInt32(),"VC#.netTrayIcon");    
		}
        
		/*
		==============================================================================
		 Method:        butChgIcon_Click
		
		 Description:  Changes the icon in system tray area.
		==============================================================================
        */
		private void butChgIcon_Click(object sender, System.EventArgs e)
		{
		  //change the icon in system tray area
          //by another one in ImageList
			Object i = 2;     
			axTrayIcon1.ChangeIcon(axImageList1.ListImages.get_Item(ref i).ExtractIcon().Handle);      
		}

		/*
		==============================================================================
		 Method:       butHide_Click
		
		 Description:  Removes the icon from system tray.
		==============================================================================
		*/
		private void butHide_Click(object sender, System.EventArgs e)
		{
		   axTrayIcon1.CtlHide(); 
		}
        
		/*
		'==================================================================================
		' Method:        butAni_Click
		'
		' Description:  Animate the icon in system tray using in icons in imagelist control.
		'==================================================================================
        */
		private void butAni_Click(object sender, System.EventArgs e)
		{   
			//Only if the icon is not Animated   
			if (!axTrayIcon1.AnimateState)  
			   axTrayIcon1.Animate(axImageList1,1);
			/*
			    Start animating the icons in system tray
                we use the icons in the imagelist control
                Then
                We pass the ImageList1 control to Animate method
                '''you can replace the icons with your own'''
            */
		}

		/*
		'==============================================================================
		' Method:       butStAni_Click
		'
		' Description:  Stop animating the icons in system tray.
		'==============================================================================
        */
		private void butStAni_Click(object sender, System.EventArgs e)
		{
		   if (axTrayIcon1.AnimateState)  
		      axTrayIcon1.StopAnimateing(this.Icon.Handle.ToInt32());    
		}

		/*
		'==============================================================================
		' Method:       butShowBalloon_Click
		'
		' Description:  Displays a balloon tooltip points to the icon.
		'==============================================================================
        */
		private void butShowBalloon_Click(object sender, System.EventArgs e)
		{
		   axTrayIcon1.ShowBalloon("VC#.NET Balloon Tooltip.","My program",ASTC.BIcoType.Info ,5000);
		}

	    /*
		'==============================================================================
		' Method:       butCBalloon_Click
		'
		' Description:  Close the balloon tooltip.
		'==============================================================================
        */ 
		private void butCBalloon_Click(object sender, System.EventArgs e)
		{
		    axTrayIcon1.CloseBalloon(); 
		}
        
		/*
		'==============================================================================
		' Method:       butTrack_Click
		'
		' Description:  Start Tracking the popup menu in system tray area.
		'==============================================================================
        */
		private void butTrack_Click(object sender, System.EventArgs e)
		{
			axTrayIcon1.TrackPopMenu = true;
			butSTrack.Enabled = true; 
			MessageBox.Show(this,"when you right click the icon a popupmenu will appear,ASTC will Track the popupmenu and close it when you left click again");  
	    }

        /*
		'==============================================================================
		' Method:       butSTrack_Click
		'
		' Description:  Stop track the popup menu in system tray area.
		'==============================================================================
        */ 
		private void butSTrack_Click(object sender, System.EventArgs e)
		{ 
			axTrayIcon1.TrackPopMenu = false;
			butSTrack.Enabled = false;
			MessageBox.Show(this, "ASTC stop tracking the popupmenu,Now right click the icon again and you will see that popup menu willn't close unless you click an item.");
	    }

        /*
		'==============================================================================
		' Method:       Form1_Closing
		'
		' Description:  Removes the icon from system tray whenever the
		'               terminates the program.
		'==============================================================================
        */
		private void Form1_Closing(object sender, System.ComponentModel.CancelEventArgs e)
		{ 
			if (axTrayIcon1.IsDisplayed)  
				axTrayIcon1.CtlHide();  
		}

        /*
		'==============================================================================
		' Method:        MenuItem2_Click
		'
		' Description:  Removes the icon from system tray and ends the program.
		'==============================================================================
        */
		private void MenuItem2_Click(object sender, System.EventArgs e)
		{   
			if (axTrayIcon1.IsDisplayed)  
				axTrayIcon1.CtlHide();  
		     this.Close(); 
		}

		private void MenuItem1_Click(object sender, System.EventArgs e)
		{
		   MessageBox.Show("main menu Item had been clicked.");  
		}
        
        /*
		'==============================================================================
		' Method:       AxTrayIcon1_BalloonLeftClick
		'
		' Description:  Display a message box whenever you left click the balloon tooltip.
		'==============================================================================
        */
		private void axTrayIcon1_BalloonLeftClick(object sender, System.EventArgs e)
		{
		   MessageBox.Show("The Balloon Tooltip had been left clicked.");  
		}
        
        /*     
		'====================================================================================
		' Method:       AxTrayIcon1_RightMouseUp
		'
		' Description:  Display a popup menu whenever you right click the icon in system tray.
		'====================================================================================
        */
		private void axTrayIcon1_RightMouseUp(object sender, System.EventArgs e)
		{
		   axTrayIcon1.PopUp(ContextMenu1.Handle.ToInt32(),this.Handle.ToInt32());  
		}
	}
}
